src/components/StudyRoom/StudyRoomSettings.js

```diff
@@ -0,0 +1,198 @@
+            description: studyRoom.description
+        })
+        setLoading(false);
+        //let email = JSON.parse(localStorage.getItem("email"));
+        // axios.get(`http://localhost:5000/room/${email}`)
+        //     .then(res => {
+        //         const data = res.data[0];
+        //         console.log(data);
+        //         setRoomData({...roomData, title: data.title,avatarText: data.avatarText,color: data.color,description: data.description});
+        //         console.log(roomData)
+        //     })
+        //     .catch(err => {console.log(`Error: ${err}`); setErrorMessage(`${err}`.substring(44) === 401 ? 'request could not be sent' : `${err}`)});
+     },[])
+
+    React.useEffect(()=>{
+        fetchData();
+        let email = JSON.parse(localStorage.getItem("email"));
+        //user needs to be logged in to access
+        if(email === undefined || email === ''){
+            navigate("/login");
+        }
+        // only an owner can access the settings page
+        if(roomData.owner !== email && roomData.owner != ''){
+            navigate("/study-room-home");
+        }
+
+    },[loading])
+
+    // sends updated field to db
+    function handleUpdate(e){
+        e.preventDefault();
+        let avatarIconText = SetAvatarText(roomData.title);
+        setRoomData({...roomData, avatarText: avatarIconText});
+        // axios.post('http://localhost:5000/room/',roomData)
+        //     .then(res => {
+        //         console.log(res);
+        //         navigate("/study-room-home");
+        //     })
+        //     .catch(err => {console.log(`Error: ${err}`); setErrorMessage(`${err}`.substring(44) === 401 ? 'request could not be sent' : `${err}`)});
+    }
+
+    //deletes the room in db
+    function handleDelete(e){
+        // axios.post('http://localhost:5000/room/delete',{username:roomData.owner, sID:roomData.sid})
+        //     .then(res => {
+        //         console.log(res);
+        //         navigate("/study-room-home");
+        //     })
+        //     .catch(err => {console.log(`Error: ${err}`); setErrorMessage(`${err}`.substring(44) === 401 ? 'request could not be sent' : `${err}`)});
+    }
+
+    function handleTitleChange(e){
+        let avatar = SetAvatarText( e.target.value)
+        setRoomData({...roomData, title: e.target.value, avatarText:avatar});
+    }
+
+    function handleColorChange(e){
+        setRoomData({...roomData, color: e.target.value});
+    }
+
+    function handleDescriptionChange(e){
+        setRoomData({...roomData, description: e.target.value});
+    }
+
+    function SetAvatarText(t){
+        let finalText='';
+        let displayText = t.split(' ').map(word => word[0]).join('');
+
+        if(displayText.length >4){
+            finalText = displayText.substring(0,4);
+        }else{
+            finalText = displayText;
+        }
+        return finalText;
+    }
+
+    const updateRoom = (
+        <React.Fragment>
```

Comment:
Can you wrap this in a div with responsive height (instead of the React.Fragment)? The issues depicted in the photos seem to be related to responsiveness.

Reply:
I tried adding a div with viewport height and tested with different window sizes and I saw no difference. Ill create a bug for this as it requires more testing to see exactly what creates these window differences. Ill close this issue and merge the branch